lib/index.js

```diff
@@ -74,6 +78,13 @@ class BaseAsset {
             return Promise.all(promises);
         }));
     }
+
+    _get_outgoing_asset_ids() {
+        if (this._thumbnail_asset_id) {
+            return [ this._thumbnail_asset_id ];
+        }
+        return [];
+    }
```

Comment:
This is a bit tricky here... since all assets types will have this code you're technically implementing `_get_thumbnail_asset_id()` which any other asset that needs to customize will need to override with `super.` invocation.

I think a better way to implement this would be to have `_get_thumbnail_asset_id` (returns thumbnail asset id if there is one), `get_dependent_assets` (that the base has, that is used by APIs, and that has an invocation of `_get_thumbnail_asset_id`), and `_get_dependent_asset_id` (that children can override and add any other non-thumbnail asset id to the list). Let me know on slack if this might not explain things well.